authorization-dotnet/TwoFactorAuthenticationGoogleAuthenticatorAngular/TwoFactorAuthenticationGoogleAuthenticatorAngular/Controllers/AccountsController.cs

```diff
@@ -0,0 +1,161 @@
+            _urlEncoder = urlEncoder;
+            _jwtHandler = jwtHandler;
+        }
+
+        [HttpPost("Registration")]
+        public async Task<IActionResult> RegisterUser([FromBody] UserForRegistrationDto userForRegistration)
+        {
+            if (userForRegistration == null || !ModelState.IsValid)
+                return BadRequest();
+
+            var user = _mapper.Map<User>(userForRegistration);
+
+            var result = await _userManager.CreateAsync(user, userForRegistration.Password);
+            if (!result.Succeeded)
+            {
+                var errors = result.Errors.Select(e => e.Description);
+
+                return BadRequest(new RegistrationResponseDto { Errors = errors });
+            }
+
+            return StatusCode(201);
+        }
+
+        [HttpPost("Login")]
+        public async Task<IActionResult> Login([FromBody] UserForAuthenticationDto userForAuthentication)
+        {
+            var user = await _userManager.FindByNameAsync(userForAuthentication.Email);
+
+            if (user == null || !await _userManager.CheckPasswordAsync(user, userForAuthentication.Password))
+                return Unauthorized(new AuthResponseDto { ErrorMessage = "Invalid Authentication" });
+
+            var isTfaEnabled = await _userManager.GetTwoFactorEnabledAsync(user);
+
+            if(!isTfaEnabled)
+            {
+                var signingCredentials = _jwtHandler.GetSigningCredentials();
+                var claims = _jwtHandler.GetClaims(user);
+                var tokenOptions = _jwtHandler.GenerateTokenOptions(signingCredentials, claims);
+                var token = new JwtSecurityTokenHandler().WriteToken(tokenOptions);
+
+                return Ok(new AuthResponseDto { IsAuthSuccessful = true, IsTfaEnabled = false, Token = token });
+            }
+
+            return Ok(new AuthResponseDto { IsAuthSuccessful = true, IsTfaEnabled = true });
+        }
+
+        [HttpPost("LoginTfa")]
+        public async Task<IActionResult> LoginTfa([FromBody] TfaDto tfaDto)
+        {
+            var user = await _userManager.FindByNameAsync(tfaDto.Email);
+
+            if (user == null)
+                return Unauthorized(new AuthResponseDto { ErrorMessage = "Invalid Authentication" });
+
+            var validVerification = await _userManager.VerifyTwoFactorTokenAsync(user, _userManager.Options.Tokens.AuthenticatorTokenProvider, tfaDto.Code);
+            if (!validVerification)
+                return BadRequest("Invalid Token Verification");
+
+            var signingCredentials = _jwtHandler.GetSigningCredentials();
+            var claims = _jwtHandler.GetClaims(user);
+            var tokenOptions = _jwtHandler.GenerateTokenOptions(signingCredentials, claims);
+            var token = new JwtSecurityTokenHandler().WriteToken(tokenOptions);
+
+            return Ok(new AuthResponseDto { IsAuthSuccessful = true, IsTfaEnabled = true, Token = token });
+        }
+
+        [HttpGet("TfaSetup")]
```

Comment:
Use the so-called "kebab-case" for naming routes, this should be "tfa-setup". Plase apply this to all routes.